### ISSUE TITLE
AssetCanisterApiClient.UploadAssetChunkedAsync - file is not chunked correctly

### ISSUE THREAD

**Issue by stefan-adna** [BoomDAO/ICP.NET#123]
When using the AssetCanisterApiClient Class to upload files there is a problem when chunking the file.

```
int bytesRead = await contentStream.ReadAsync(buffer.AsMemory());
...
byte[] chunkBytes = bytesRead < buffer.Length
				? buffer[0..(bytesRead - 1)]
				: buffer;
```
The upper bound of the array selector is exclusive, meaning reading 5 bytes should be `buffer[0..5]` which reads position 0-4 therefore the correct code should be `buffer[0..(bytesRead - 1)]` (without the -1).

**Comment by Gekctek** [comment]
Nice catch, ill fix

**Comment by Gekctek** [comment]
Fixed in 6.1.0

**Comment by stefan-adna** [comment]
Very nice. Thanks a lot 👍
